packages/core/src/formula/formula.ts

```diff
@@ -14,7 +14,22 @@ import { isToddleFormula } from './formulaTypes'
 declare const document: Document | undefined
 type ShadowRoot = DocumentFragment
 
-export interface PathOperation extends ToddleMetadata {
+interface BaseOperation extends ToddleMetadata {
+  type:
+    | 'path'
+    | 'function'
+    | 'object'
+    | 'array'
+    | 'record'
+    | 'value'
+    | 'apply'
+    | 'or'
+    | 'and'
+    | 'switch'
```

Comment:
These are all added in the individual types below and allow us to work with discriminated unions. Why should they also be added here?
```suggestion
```